bibliopixel/drivers/spi_driver_base.py

```diff
@@ -1,75 +1,158 @@
-import os, sys, time
+import fcntl
+import math
+import os
+import struct
 from . channel_order import ChannelOrder
 from . driver_base import DriverBase
 from .. import log
 
 
-class DriverSPIBase(DriverBase):
-    """
-    Base driver for controling SPI devices on systems like the Raspberry Pi and
-    BeagleBone.
-    """
-    def __init__(self, num, c_order=ChannelOrder.GRB, use_py_spi=True,
-                 dev='/dev/spidev0.0', SPISpeed=2, open=open, gamma=None):
-        super().__init__(num, c_order=c_order, gamma=gamma)
+class SpiBaseInterface(object):
+    """ abstract class for different spi backends"""
 
-        self.dev = dev
-        self.use_py_spi = use_py_spi
-        self._spiSpeed = SPISpeed
-        self._open = open
+    def __init__(self, dev, SPISpeed):
```

Comment:
changed.
Also done this for the driver to have python conform variable names.
https://www.python.org/dev/peps/pep-0008/#id47